File: LogiOS/TrustarApp/Repo/Data/BinDetailData.swift
//
//  BinDetailData.swift
//  TrustarApp
//
//  Created by CuongNguyen on 03/03/2022.
//  Copyright © 2022 DionSoftware. All rights reserved.
//

import Foundation

class BinDetailData {
    
    var binDetail: BinDetail
    var delayReasons: [DelayReason]
    var incidentalHeaderCount: Int
    var signedIncidentalHeaderCount: Int
    
    init(binDetail: BinDetail, delayReasons: [DelayReason], incidentalHeaderCount: Int, signedIncidentalHeaderCount: Int) {
        self.binDetail = binDetail
        self.delayReasons = delayReasons
        self.incidentalHeaderCount = incidentalHeaderCount
        self.signedIncidentalHeaderCount = signedIncidentalHeaderCount
    }
    
    var delayReason: DelayReason? {
        if let cd = binDetail.delayReasonCd {
           return delayReasons.first(where: { $0.reasonCd == cd })
        }
        
        return nil
    }
    
    var incidentalEnable: Bool {
        return Current.Shared.loggedUser?.userInfo.incidentalEnable() == true
    }
}

File: LogAndroid/TrustarApp/src/main/kotlin/jp/co/toukei/log/trustar/other/BinDetailData.kt
package jp.co.toukei.log.trustar.other

import jp.co.toukei.log.trustar.Current
import jp.co.toukei.log.trustar.db.user.entity.BinDetail
import jp.co.toukei.log.trustar.db.user.entity.DelayReason
import jp.co.toukei.log.trustar.db.user.entity.Work

class BinDetailData(
    @JvmField val binDetail: BinDetail,
    @JvmField val delayReasons: Array<DelayReason>,
    @JvmField val incidentalHeaderCount: Int,
    @JvmField val signedIncidentalHeaderCount: Int,
    @JvmField val work: Work?,
) {
    val delayReason: DelayReason? = binDetail.delayReasonCd?.let { cd ->
        delayReasons.firstOrNull { it.reasonCd == cd }
    }

    @JvmField
    val incidentalEnabled = Current.user?.userInfo?.incidentalEnabled == true
}
